src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -445,7 +445,17 @@ else if (kruizeObject.getMode().equalsIgnoreCase(AnalyzerConstants.AUTO) || krui
                         experimentName, interval_start_time, interval_end_time));
                 kruizeObject.setValidation_data(new ValidationOutputData(false, e.getMessage(), HttpServletResponse.SC_INTERNAL_SERVER_ERROR));
             }
-        } catch (Exception | FetchMetricsError | InvalidModelException | InvalidTermException e) {
+        } catch (FetchMetricsError e) {
+            LOGGER.error(String.format(AnalyzerErrorConstants.APIErrors.UpdateRecommendationsAPI.RECOMMENDATION_EXCEPTION,
```

Comment:
The `RECOMMENDATION_EXCEPTION` constant contains `%s` to format the String using the variables passed but in case of LOGGER, you need to put `{}` instead.

For e.g:
           ` RECOMMENDATION_EXCEPTION = "Exception occurred while generating recommendations for experiment: {} and interval_end_time: {} : {} "; `